From akpm@osdl.org Tue Jan 17 15:37:25 2006
Message-Id: <200601172337.k0HNbPhc006451@shell0.pdx.osdl.net>
From: Alan Cox <alan@lxorguk.ukuu.org.uk>
Subject: USB: libusual: fix warning on 64bit boxes
To: alan@lxorguk.ukuu.org.uk, alan@redhat.com, greg@kroah.com,
   zaitcev@redhat.com, mm-commits@vger.kernel.org
From: akpm@osdl.org
Date: Tue, 17 Jan 2006 15:39:25 -0800

From: Alan Cox <alan@lxorguk.ukuu.org.uk>

We cast an int to a void * which not unreasonably makes gcc suspicious.
We don't actually care what type "type" is so use unsigned long so it
matches pointer length on all platforms.

Signed-off-by: Alan Cox <alan@redhat.com>
Acked-by: Pete Zaitcev <zaitcev@redhat.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/storage/libusual.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/storage/libusual.c
+++ gregkh-2.6/drivers/usb/storage/libusual.c
@@ -116,7 +116,7 @@ EXPORT_SYMBOL_GPL(usb_usual_check_type);
 static int usu_probe(struct usb_interface *intf,
 			 const struct usb_device_id *id)
 {
-	int type;
+	unsigned long type;
 	int rc;
 	unsigned long flags;
 
